### PR TITLE
add iswap and dcx gates

### PR DESCRIPTION
**Pull request by ajavadia**
adding two new gates for iswap and double-cnot.

### CODE REVIEW

File: qiskit/extensions/standard/iswap.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    This gate is symmetric.
+    """
+
+    def __init__(self):
+        super().__init__('iswap', 2, [])
+
+    def _define(self):
+        """
+        gate iswap a,b {
+            u2(pi/2) a;
```
Comment (kdk) [inline qiskit/extensions/standard/iswap.py]:
```suggestion
            u2(pi/2, 0) a;
```

File: qiskit/extensions/standard/iswap.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            u2(-pi/2) b;
```
Comment (kdk) [inline qiskit/extensions/standard/iswap.py]:
```suggestion
            u2(-pi/2, 0) b;
```

**Comment by ajavadia** [comment]
no idea what the doc error means here
```
home/vsts/work/1/s/qiskit/extensions/standard/iswap.py:docstring of qiskit.extensions.iSwapGate:3:Inline substitution_reference start-string without end-string.
```

File: qiskit/extensions/standard/iswap.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import QuantumRegister
+
+
+class iSwapGate(Gate):
+    r"""iSWAP gate.
+
+    A 2-qubit XY interaction that is equivalent to a SWAP up to a diagonal.
+    This is a Clifford and symmetric gate. It's action is to swap two qubit
+    states and phase the |01⟩ and |10⟩ amplitudes by i.
```
Comment (mtreinish) [inline qiskit/extensions/standard/iswap.py]: This should fix that doc error:

```suggestion
    states and phase the :math:`|01\\rangle` and :math:`|10\\rangle` amplitudes by i.
```

Comment (ajavadia) [reply qiskit/extensions/standard/iswap.py]: oh i see let me try this thanks!

**Pull request merged**